src/gwt/src/org/rstudio/studio/client/common/rstudioapi/DialogHtmlSanitizer.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * DialogHtmlSanitizer.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+package org.rstudio.studio.client.common.rstudioapi;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.google.gwt.safehtml.shared.HtmlSanitizer;
+import com.google.gwt.safehtml.shared.SafeHtml;
+import com.google.gwt.safehtml.shared.SafeHtmlUtils;
+
+public final class DialogHtmlSanitizer implements HtmlSanitizer {
+   private static final DialogHtmlSanitizer INSTANCE = new DialogHtmlSanitizer();
+
+   private static final Set<String> TAG_WHITELIST = new HashSet<String>(
+      Arrays.asList(
+         "p", "em", "strong", "b", "i"
+      )
+   );
+
+   public static DialogHtmlSanitizer getInstance() {
+      return INSTANCE;
+   }
+
+   public static SafeHtml sanitizeHtml(String html) {
+      if (html == null) {
+         html = "";
+      }
+      
+      return SafeHtmlUtils.fromTrustedString(dialogSanitize(html));
+   }
+
+   private static String dialogSanitize(String text) {
+      StringBuilder sanitized = new StringBuilder();
+
+      boolean firstSegment = true;
+      for (String segment : text.split("<", -1)) {
+         if (firstSegment) {
+            firstSegment = false;
+            sanitized.append(SafeHtmlUtils.htmlEscapeAllowEntities(segment));
+            continue;
+         }
+
+         int tagStart = 0;
+         int tagEnd = segment.indexOf('>');
+         String tag = null;
+         boolean isValidTag = false;
+         if (tagEnd > 0) {
+            if (segment.charAt(0) == '/') {
+               tagStart = 1;
+            }
+            tag = segment.substring(tagStart, tagEnd);
```

Comment:
Should this get `tolower`'ed so that `<P>`, etc. can be used?

As written it looks like this will also reject tags with attributes -- this is the safest stance (as we wouldn't want e.g. `<p onmouseover="pwn();">`) but we might also consider a whitelist of attributes in the future.